src/dotnet-svcutil/lib/src/CommandProcessorOptions.cs

```diff
@@ -627,7 +627,12 @@ private async Task ProcessReferencesOptionAsync(CancellationToken cancellationTo
                     {
                         for (int idx = this.References.Count - 1; idx >= 0; idx--)
                         {
-                            if (!references.Contains(this.References[idx]))
```

Comment:
Clarification for the change:

The purpose of this code piece is to remove any reference that exist in References but not exist in references (as some reference could have been removed after service was added) when updating service.
`References `is the original reference list
`references `is the resolved reference list of the running project

With the fix of this issue applied, the reference "name" has changed from full local path to assembly name, so the old reference item in References collection might not exactly equals to the reference in references collection due to the path change. To resolve this discrepancy, 
1. the containment check previously done with `(if(references.Contains(this.References[idx])))` is now performed by comparing the reference item's Name property for equality.
2. since only reference items present in the references list can be used for type reuse (as they should be available or resolved from the running project), we need to update the existing Reference (which could be with a local full path) with the reference item from the resolved references list. The assignment `this.References[idx] = selected;` does not add a new item to the list but updates the existing item in References with the corresponding reference (having the same name) from the resolved project references.
3. traversing the collection using array index ensures the shape of the References collection is maintained during the removal and update operations.